source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1360,45 +1360,90 @@ then a re-check will be triggered by the server selection algorithm.
 
 "not master" and "node is recovering"
 `````````````````````````````````````
-
 These errors are detected from a getLastError response,
-write command response, or query response. Clients MUST consider a server
-error to be a "node is recovering" error if the substrings "node is recovering"
-or "not master or secondary" are anywhere in the error message.
-Otherwise, if the substring "not master" is in the error message it is a
-"not master" error::
+write command response, or query response. Clients MUST check if the server
+error is a "node is recovering" error or a "not master" error.
+
+If the response includes an error code, it MUST be checked first to attempt
+to determine if error is a "node is recovering" or "not master" error.
+
+The following error codes indicate a replica set member is temporarily
+unusable. These are called "node is recovering" errors:
+
+.. list-table::
+  :header-rows: 1
+
+  * - Error Name
+    - Error Code
+  * - InterruptedAtShutdown
+    - 11600
+  * - InterruptedDueToReplStateChange
```

Comment:
I'm don't have a super firm grasp on how the drivers treat "node is recovering" vs "not master" errors, but this happens when a command is interrupted by a node transitioning from primary to secondary, that makes me think it should be a 'not master' error

Reply:
Maybe the names aren't great, but "not master" errors indicate the server's role is something that SDAM can use. If an InterruptedDueToReplStateChange error is returned, does that mean the node is finished transitioning, or still in the process?

Reply:
it may or may not have finished, it certainly could still be processing the stepdown.

Reply:
The only difference between "not master" and "node is recovering" is in single trheaded drivers:

> Multi-threaded and asynchronous clients MUST request an immediate check of the server. Unlike in the "network error" scenario above, a "not master" or "node is recovering" error means the server is available but the client is wrong about its type, thus an immediate re-check is likely to provide useful information.
> 
> For single-threaded clients, in the case of a "not master" error, the client MUST check the server immediately (see checking a single server). For a "node is recovering" error, single-threaded clients MUST NOT check the server, as an immediate server check is unlikely to find a usable server.

So I think it is fine to keep PrimarySteppedDown and PrimarySteppedDown as "node is recovering".